invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/CreatibutorsField/CreatibutorsModal.js

```diff
@@ -155,21 +171,16 @@ export class CreatibutorsModal extends Component {
     if (serializeCreatibutor) {
       return serializeCreatibutor(submittedCreatibutor, initialCreatibutor);
     }
-    const findField = (arrayField, key, value) => {
-      const knownField = _find(arrayField, {
-        [key]: value,
-      });
-      return knownField ? knownField : { [key]: value };
-    };
+
     const identifiersFieldPath = "person_or_org.identifiers";
     const affiliationsFieldPath = "affiliations";
     // The modal is saving only identifiers values, thus
     // identifiers with existing scheme are trimmed
-    // Here we merge back the known scheme for the submitted identifiers
     const initialIdentifiers = _get(initialCreatibutor, identifiersFieldPath, []);
     const submittedIdentifiers = _get(submittedCreatibutor, identifiersFieldPath, []);
     const identifiers = submittedIdentifiers.map((identifier) => {
-      return findField(initialIdentifiers, "identifier", identifier);
+      // Here we merge back the known scheme for the submitted identifiers
+      return this.findField(initialIdentifiers, "identifier", identifier);
```

Comment:
```suggestion
      return this.findField(this.state.personIdentifiers, "identifier", identifier);
```
what about just keep the `findField` generic enough and compute the logic of fallbacks in the serialize function?